Can't build on vs2022 17.10 msvc on latest code.

```
template<typename T>
constexpr decltype(auto) delay_sp_config_eval() {
  if constexpr (sizeof(T)==0) {
    return (T*){};
  }
  else {
    return (sp_config*){};
  }
}

yalantinglibs\include\ylt\struct_pack\reflection.hpp(596,16): error C4576: a parenthesized type followed by an initializer list is a non-standard explicit type conversion syntax
yalantinglibs\include\ylt\struct_pack\reflection.hpp(599,24): error C4576: a parenthesized type followed by an initializer list is a non-standard explicit type conversion syntax
yalantinglibs\include\ylt\struct_pack\reflection.hpp(596,18): error C2760: syntax error: ';' was unexpected here; expected '}'
yalantinglibs\include\ylt\struct_pack\reflection.hpp(597,3): error C2760: syntax error: '}' was unexpected here; expected ';'
yalantinglibs\include\ylt\struct_pack\reflection.hpp(599,26): error C2760: syntax error: ';' was unexpected here; expected '}'
yalantinglibs\include\ylt\struct_pack\reflection.hpp(600,3): error C2760: syntax error: '}' was unexpected here; expected ';'
```

--- a/include/ylt/struct_pack/reflection.hpp
+++ b/include/ylt/struct_pack/reflection.hpp
@@ -593,10 +593,10 @@ template <typename T, typename = void>
 template<typename T>
 constexpr decltype(auto) delay_sp_config_eval() {
   if constexpr (sizeof(T)==0) {
-    return (T*){};
+    return (T*)nullptr;
   }
   else {
-    return (sp_config*){};
+    return (sp_config*)nullptr;
   }
 }
 

